raiden/network/rpc/client.py

```diff
@@ -293,29 +306,35 @@ class BlockChainService(object):
     """ Exposes the blockchain's state through JSON-RPC. """
     # pylint: disable=too-many-instance-attributes
 
-    def __init__(
-            self,
-            privatekey_bin,
-            registry_address,
-            jsonrpc_client,
-            poll_timeout=DEFAULT_POLL_TIMEOUT):
-
+    def __init__(self, privatekey_bin, jsonrpc_client, poll_timeout=DEFAULT_POLL_TIMEOUT):
         self.address_to_token = dict()
         self.address_to_discovery = dict()
-        self.address_to_channelmanager = dict()
         self.address_to_nettingchannel = dict()
         self.address_to_registry = dict()
-        self.token_to_channelmanager = dict()
 
         self.client = jsonrpc_client
         self.private_key = privatekey_bin
         self.node_address = privatekey_to_address(privatekey_bin)
         self.poll_timeout = poll_timeout
-        self.default_registry = self.registry(registry_address)
 
     def block_number(self):
         return self.client.blocknumber()
 
+    def is_synced(self):
+        result = self.client.call('eth_syncing')
+
+        # the node is synchronized
+        if result is False:
```

Comment:
I noticed you still rely on this function if the oracle call fails for some reason.

I would suggest to either modify  `is_synced()` a bit or totally remove it. You can test it yourself and see why `eth_syncing()` is not reliable at all if used on its own. 

For example right now I am syncing `geth --fast --testnet console` for ropsten.

With `eth.syncing` I get:

```                                                                                                                                                                                                                                                            {                                                                                                                                                                                                                                                                               
  currentBlock: 1695773,
  highestBlock: 1695987,
  knownStates: 15536487,
  pulledStates: 15532242,
  startingBlock: 0
}
```

but 

`eth.blockNumber` is still at `0` because the client is still importing state entries, not even close to being finished.

So if you use `currentBlock` and `highestBlock` from `eth.syncing` as you do in this logic, raiden will think that the sync finished. 

But you can modify the logic of the function and use the `highestBlock` from `eth.syncing` and the `currentBlock` as returned from `eth.blockNumber`. Then it will be quite a bit more accurate I believe.